crest/Assets/Crest/Crest/Scripts/Shapes/FFT/ShapeFFT.cs

```diff
@@ -108,7 +152,15 @@ public partial class ShapeFFT : MonoBehaviour, LodDataMgrAnimWaves.IShapeUpdatab
 
         internal float LoopPeriod => _enableBakedCollision ? _timeLoopLength : -1f;
 
-        Mesh _meshForDrawingWaves;
+        public IPaintedData PaintedData => _paintData;
+        public Shader PaintedInputShader => null;
+
+        Mesh MeshForGeneration => _inputMode == Mode.Spline ? _meshForDrawingWavesSpline : (_inputMode == Mode.CustomGeometryAndShader ? _meshForDrawingWavesCustom : null);
+        bool UsingGeometryToGenerate => _inputMode == Mode.Spline || _inputMode == Mode.CustomGeometryAndShader;
+
+        // Will either be spline mesh, or custom geo assigned by user
+        Mesh _meshForDrawingWavesSpline;
+        Mesh _meshForDrawingWavesCustom;
```

Comment:
You couldnt get custom geo to work with FFT? Hm i have not tried that in forever. I'll look around

Reply:
A tempting fallback will be to remove this. I highly doubt anyone will use it over painting/spline. I guess legacy data is a consideration - not sure i heard of anyone using it.

Reply:
Ok funny enough, basic validation was basically missing for FFT. Added that first.

Also made it actually initialise *_meshForDrawingWavesCustom*

With the benefit of hindsight i think FFT could have been implemented as *RegisterFFTInput* and be deeply connected into all the stuff in RegisterLodInput. It borrows parts of it slightly awkwardly. But this would entail a migration nightmare i guess - and a rewrite.

I think this works as is but lmk what you think

Reply:
Oh... wait... I just remembered that I don't think FFTs supported renderers as it was all in on splines. Apologies.

Reply:
I believe removing it is the best option, and I've done so now

Reply:
Also I retested on master and it doesnt work there either. Not sure why i left that code hanging around. I have therefore not bothered to add RN.